libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1314,7 +1314,7 @@ MAV_RESULT GCS_MAVLINK::_set_mode_common(const MAV_MODE base_mode, const uint32_
         if (set_mode(custom_mode)) {
             result = MAV_RESULT_ACCEPTED;
         }
-    } else if (base_mode == MAV_MODE_FLAG_DECODE_POSITION_SAFETY) {
+    } else if (base_mode == (MAV_MODE)MAV_MODE_FLAG_SAFETY_ARMED) {
```

Comment:
I understand your point, but I would change it to be `(base_mode & MAV_MODE_FLAG_DECODE_POSITION_SAFETY)`. The reason is that the MAV_MODE is actually saying to arm the vehicle, but we depend on custom mode to say if it is to arm or disarm, so I think that using the flag that says we should decode the "position safety" is better (although the name is also bad).
Also, the MAVLink spec is very specific in saying that MAV_MODE_FLAG_SAFETY_ARMED should be ignored in MAV_CMD_DO_SET_MODE.

In reality I don't know why we have two enums for exactly the same, but the decode ones seem better for us since we then use custom mode to know what to do.

Reply:
OK, I'm reverting this to be the DECODE_POSITION stuff.  Incidentally, my theory is that this was supposed to be some sort of indirection, and the values might have been bit offsets rather than bit masks.

In terms of using `&` - I don't think that can work, because we'd then be inconsistent with the comment on the following line, which says that the safety switch thing has to be in a "command of its own".

Reply:
> In terms of using & - I don't think that can work, because we'd then be inconsistent with the comment on the following line, which says that the safety switch thing has to be in a "command of its own".

I thought of that too, but I thought it wouldn't matter because we only support setting mode or arming switch state and, with the use of `else if`, we only allow one or the other. It would still allow some GCS to try and send other stuff in the same packet, but we don't support it anyway.